### ISSUE TITLE
Yarn resolutions

### ISSUE THREAD

**Issue by ryanto** [ember-cli/ember-try#208]
I'm working on an addon that adds functionality to Ember Data and I've found myself in a situation where the wrong version of data ends up in my final build. My ember-try config looks something like this...

```js
// ...
{
  name: 'ember-lts-2.12',
  npm: {
    devDependencies: {
      'ember-source': '~2.12.0',
      'ember-data': '~2.12.0'
    }
  }
},
// ...
{
  name: 'ember-3.1',
  npm: {
    devDependencies: {
      'ember-source': '~3.1.0',
      'ember-data': '~3.1.0'
    }
  }
},
// ...
```

And this was working great! But then I installed addon docs, which has a direct dependency on ember data: https://github.com/ember-learn/ember-cli-addon-docs/blob/v0.5.3/package.json#L48

When my Ember application is built, my build ends up with the addon docs version of Ember data. So, my tests get run against data version 2.18, regardless of what's in my try config. Yikes!

Yarn has the option to supply [resolutions](https://yarnpkg.com/en/docs/selective-version-resolutions) for overriding a dependency of a dependency. This would allow my addon to override addon docs data dependency. I tried this, and it works on the command line, but it looks like ember-try does not support the resolutions hash. This was my config...

```js
{
  name: 'ember-lts-2.12',
  npm: {
    devDependencies: {
      'ember-source': '~2.12.0',
      'ember-data': '~2.12.0'
    },
    resolutions: {
      'ember-data': '~2.12.0'
    }
  }
},
```

But the `resolutions` did not end up in my package.json when running `ember try`.

I wanted to see if you would be open to having ember-try support resolutions?

PS: I opened an issue for [addon docs](https://github.com/ember-learn/ember-cli-addon-docs/issues/242) as well.

**Comment by samselikoff** [comment]
@kategengler We're running into this again. Any chance you'd support adding this to Ember Try? We might be able to work on a PR

**Comment by kategengler** [comment]
ping @rwjblue for thoughts -- I think we had discussed this in person but I sadly have forgotten all details

**Comment by rwjblue** [comment]
I think its probably fine to support, but instead of adding custom support for `resolutions` we should consider if we should deeply merge _any_ key with `package.json` contents...

**Comment by ryanto** [comment]
I think deeply merging all keys is good. If Sam and I were to try this, is the best place to start here: https://github.com/ember-cli/ember-try/blob/master/lib/dependency-manager-adapters/npm.js

Also, I'll finally I'll be able to test this scenario!

```js
scenarios: [{
  name: 'Magically fix all bugs',
  npm: {
    contributors: ["Robert Jackson"]
  }
}]
```

**Comment by kategengler** [comment]
Deeply merging makes sense to me, but I wonder if special support for resolutions would also make sense? In the way we did it for bower -- if using yarn, if a specific dep is specified add the package also to resolutions.

**Comment by ryanto** [comment]
Great Katie, that made it a lot easier.

I opened a PR here: https://github.com/ember-cli/ember-try/pull/231

**Comment by kategengler** [comment]
Released with #231 in v1.2.0
